### ISSUE TITLE
Microsoft.Extensions.Logging.Abstractions : Error Version

### ISSUE THREAD

**Issue by Rlooping** [robinrodricks/FluentFTP#956]
**FTP OS: Unix

**FTP Server:** Pure-FTPd

**Computer OS:** Windows 11

**FluentFTP Version:** 40.0.0

**Framework:** .NET 7.4.2

Dependance : Microsoft.Extensions.Logging.Abstractions 2.1.0 Needed
But not working. Windows Need Microsoft.Extensions.Logging.Abstractions 6.0.2

```

**Comment by FanDjango** [comment]
Can you show me the error message describing the missing dependency?

**Comment by robinrodricks** [comment]
@Rlooping  I don't understand. Are you saying the default setup (2.1.0) does not work, or are you saying that you are trying a newer version (6x) which does not work?

**Comment by Rlooping** [comment]
When upgrading to version 40.0, the "Microsoft.Extensions.Logging.Abstractions" dependency was also installed.
Then Visual Studio offers an update of this dependency to version 6.0.2.
With version 6.0.2, the project launches, but transferring files causes the application to crash.
So I went back to version 2.1.0, but there, I have a version error and the project no longer works.
With version 39 of Fluent FTP, I had no problems with my project.

Error Generated with version 6.0.2 of "Microsoft.Extensions.Logging.Abstractions":
Could not load file or assembly 'Microsoft.Extensions.Logging.Abstractions, Version=2.1.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60' or one of its dependencies. The assembly manifest definition found does not match the assembly reference. (Exception from HRESULT: 0x80131040)

**Comment by FanDjango** [comment]
> With version 39 of Fluent FTP, I had no problems with my project.
Ok
> When upgrading to version 40.0, the "Microsoft.Extensions.Logging.Abstractions" dependency was also installed.
That was to be expected. And it installed 2.1.0, I expect.
> Then Visual Studio offers an update of this dependency to version 6.0.2.
Ok. It would have been better to not accept that offer.
Start from fresh, don't accept the offer to upgrade, and verify that it works with 2.10. We'll take it from there.

**Comment by Rlooping** [comment]
hello, 

The program works again with not updating "Microsoft.Extensions.Logging.Abstractions".

But I have this message at the project level:

Severity Code Description Project File Line Deletion Status
Warning detection of unresolved conflicts between different versions of "Microsoft.Extensions.Logging.Abstractions".
Conflict between "Microsoft.Extensions.Logging.Abstractions, Version=2.1.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60" and "Microsoft.Extensions.Logging.Abstractions, Version=6.0.0.2, Culture=neutral, PublicKeyToken=adb9793829ddae60" .
    "Microsoft.Extensions.Logging.Abstractions, Version=2.1.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60" was selected because it was primary type and "Microsoft.Extensions.Logging.Abstractions, Version=6.0.0.2, Culture =neutral, PublicKeyToken=adb9793829ddae60" was not.
    References that depend on "Microsoft.Extensions.Logging.Abstractions, Version=2.1.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60" [Z:\...\packages\Microsoft.Extensions. Logging.Abstractions.2.1.0\lib\netstandard2.0\Microsoft.Extensions.Logging.Abstractions.dll].
        Z:\...\... TimeClock\Pointeuse\packages\Microsoft.Extensions.Logging.Abstractions.2.1.0\lib\netstandard2.0\Microsoft.Extensions.Logging.Abstractions.dll
          Item Include of a project file that generated the reference "Z:\...\packages\Microsoft.Extensions.Logging.Abstractions.2.1.0\lib\netstandard2.0\Microsoft. Extensions.Logging.Abstractions.dll".
            Microsoft.Extensions.Logging.Abstractions, Version=2.1.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60, processorArchitecture=MSIL
    References that depend on "Microsoft.Extensions.Logging.Abstractions, Version=6.0.0.2, Culture=neutral, PublicKeyToken=adb9793829ddae60" [].
        Z:\....\packages\FluentFTP.40.0.0\lib\net472\FluentFTP.dll
          Item Include of a project file that generated the reference "Z:\....\FluentFTP.dll".
            FluentFTP, Version=40.0.0.0, Culture=neutral, PublicKeyToken=f4af092b1d8df44f, processorArchitecture=MSIL Pointer

**Comment by FanDjango** [comment]
Examine your .proj files: Perhaps you still have remnants of 6.0.0.2 in there?

**Comment by Rlooping** [comment]
Thank you, indeed there were errors in the "app.config" with a remnant of the 6.002 update.

Will this dependency be updated in the next update?

**Comment by FanDjango** [comment]
Just want to inform you: 

1. I have an application that uses FluentFTP. I have Nuget installed "Microsoft.Extensions.Logging.Abstractions" 6.0.1 which works fine. I upgraded it to 6.0.2 just now, works fine.

2. I have downloaded master from GitHub and opened it in VS2022, can build it. The "Microsoft.Extensions.Logging.Abstractions" is 2.0.1. I am offered to upgrade it to 6.0.2, which I did, and then I was able to build it and to run the tests.

No error messages, no problems. So something must still be unclean on your side. Make a backup of your current solution, and try the upgrade to 6.0.2 now. See if it ok now. Also stop/restart VS in between trys to cleanly reload the solution, sometimes it glitches there.
I am getting the offer to upgrade "Microsoft.Extensions.Logging.Abstractions" also, 2.1.0 -> 6.0.2.

**Comment by FanDjango** [comment]
Now I am also getting, as soon as I call a FluentFtp API:
```
Die Datei oder Assembly "Microsoft.Extensions.Logging.Abstractions, Version=6.0.0.2, Culture=neutral, PublicKeyToken=adb9793829ddae60" oder eine Abhängigkeit davon wurde nicht gefunden. Die gefundene Manifestdefinition der Assembly stimmt nicht mit dem Assemblyverweis überein. (Ausnahme von HRESULT: 0x80131040)
```
The Build process was ok. Calling the API caused this failure. Now that I can reproduce this, we will analyze...

**Comment by FanDjango** [comment]
> I have an application that uses FluentFTP. I have Nuget installed "Microsoft.Extensions.Logging.Abstractions" 6.0.1 which works fine. I upgraded it to 6.0.2 just now, works fine.

**Oops my bad.** Once I call a FluentFTP API function I get an error.

Here is one more piece of information:
The Nuget FluentFTP package refers to "Microsoft.Extensions.Logging.Abstractions" is 2.0.1. So in your app, you should also take this one.
Once I downloaded the code of FluentFTP from master to build it myself, I could choose to use 2.0.1, but also I could immediately update to 6.0.2. In order to avoid an error, I then **also** needed to update my reference in my app to 6.0.2, they both need to be the same.

Now, let's ask @robinrodricks if and when he might upgrade MELA to 6.0.2 and also, what are the implications of this for the different targets that are supported by FluentFTP. Might not be possible to support all, but that goes beyond my knowledge.

But for you, using the Nuget Package (right?) you need to stick with MELA 2.0.1 for now.

**Comment by jnyrup** [comment]
Here's an example of a net472 console application that uses FluentFtp 40.0.0 and Microsoft.Extensions.Logging.Abstractions 6.0.2.

[ConsoleApp67.zip](https://github.com/robinrodricks/FluentFTP/files/9640737/ConsoleApp67.zip)

The error snippet you're including suggests to me that you're not using an SDK style project like the one I've included and having problems with binding redirects, i.e. your csproj tells the .NET runtime to look for assembly version 2.1.0.0 of `Microsoft.Extensions.Logging.Abstractions.dll`, but you're including 6.0.0.2.

Be aware that assembly versions 2.1.0.0 and 6.0.0.2 corresponds to package versions 2.1.0 and 6.0.2, respectively.

**Comment by robinrodricks** [comment]
@jnyrup  Any idea why this is occuring? Afaik it was perfectly possible for a package to depend on an older nuget package, and if the user's app uses a newer one it should work perfectly right?

**Comment by jnyrup** [comment]
My example application also works when _not_ upgrading `Microsoft.Extensions.Logging.Abstractions`.

These kinds of problems are commonly referred to as [Dependency Hell](https://en.wikipedia.org/wiki/Dependency_hell).

Using SDK style project alleviates a bunch of these problems, but cannot resolve all of them, e.g. the "Diamond dependency" (see the other link).

My best advice is if possible to upgrade to SDK style projects.

Otherwise look in `app.config` to see if it contains
```
<dependentAssembly>
  <assemblyIdentity name="Microsoft.Extensions.Logging.Abstractions" publicKeyToken="adb9793829ddae60" culture="neutral" />
  <bindingRedirect oldVersion="0.0.0.0-2.1.0.0" newVersion="2.1.0.0" />
</dependentAssembly>
```

and change it to this to guide the .NET runtime to look for the newer `Microsoft.Extensions.Logging.Abstractions.dll`.
```
<dependentAssembly>
  <assemblyIdentity name="Microsoft.Extensions.Logging.Abstractions" publicKeyToken="adb9793829ddae60" culture="neutral" />
  <bindingRedirect oldVersion="0.0.0.0-6.0.0.2" newVersion="6.0.0.2" />
</dependentAssembly>
```

**Comment by robinrodricks** [comment]
@jnyrup  What is SDK style projects?

**Comment by jnyrup** [comment]
The modernized/new style of csproj that consolidates `packages.config` and many parts of `app.config` into csproj.

<details>
<summary>old style csproj</summary>

```csproj
<?xml version="1.0" encoding="utf-8"?>
<Project ToolsVersion="4.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
  <Import Project="$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props" Condition="Exists('$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props')" />
  <PropertyGroup>
    <Configuration Condition=" '$(Configuration)' == '' ">Debug</Configuration>
    <Platform Condition=" '$(Platform)' == '' ">AnyCPU</Platform>
    <ProjectGuid>{7CA92B8A-5197-4FF7-94C3-D188B06B7694}</ProjectGuid>
    <OutputType>Exe</OutputType>
    <AppDesignerFolder>Properties</AppDesignerFolder>
    <RootNamespace>ConsoleApp68</RootNamespace>
    <AssemblyName>ConsoleApp68</AssemblyName>
    <TargetFrameworkVersion>v4.7.2</TargetFrameworkVersion>
    <FileAlignment>512</FileAlignment>
    <TargetFrameworkProfile />
    <NuGetPackageImportStamp>
    </NuGetPackageImportStamp>
  </PropertyGroup>
  <PropertyGroup>
    <DebugSymbols>true</DebugSymbols>
    <DebugType>full</DebugType>
    <Optimize>false</Optimize>
    <OutputPath>bin\Debug\</OutputPath>
    <DefineConstants>TRACE;DEBUG</DefineConstants>
    <ErrorReport>prompt</ErrorReport>
    <WarningLevel>4</WarningLevel>
    <Prefer32Bit>false</Prefer32Bit>
    <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
    <TreatWarningsAsErrors>false</TreatWarningsAsErrors>
  </PropertyGroup>
  <ItemGroup>
    <Compile Include="Program.cs" />
  </ItemGroup>
  <ItemGroup>
    <Reference Include="FluentFTP, Version=40.0.0.0, Culture=neutral, PublicKeyToken=f4af092b1d8df44f, processorArchitecture=MSIL">
      <HintPath>..\packages\FluentFTP.40.0.0\lib\net472\FluentFTP.dll</HintPath>
    </Reference>
    <Reference Include="Microsoft.Bcl.AsyncInterfaces, Version=1.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51, processorArchitecture=MSIL">
      <HintPath>..\packages\Microsoft.Bcl.AsyncInterfaces.1.1.1\lib\net461\Microsoft.Bcl.AsyncInterfaces.dll</HintPath>
    </Reference>
    <Reference Include="Microsoft.CSharp" />
    <Reference Include="Microsoft.Extensions.Configuration.Abstractions, Version=3.1.8.0, Culture=neutral, PublicKeyToken=adb9793829ddae60, processorArchitecture=MSIL">
      <HintPath>..\packages\Microsoft.Extensions.Configuration.Abstractions.3.1.8\lib\netstandard2.0\Microsoft.Extensions.Configuration.Abstractions.dll</HintPath>
    </Reference>
    <Reference Include="Microsoft.Extensions.DependencyInjection.Abstractions, Version=3.1.8.0, Culture=neutral, PublicKeyToken=adb9793829ddae60, processorArchitecture=MSIL">
      <HintPath>..\packages\Microsoft.Extensions.DependencyInjection.Abstractions.3.1.8\lib\netstandard2.0\Microsoft.Extensions.DependencyInjection.Abstractions.dll</HintPath>
    </Reference>
    <Reference Include="Microsoft.Extensions.FileProviders.Abstractions, Version=3.1.8.0, Culture=neutral, PublicKeyToken=adb9793829ddae60, processorArchitecture=MSIL">
      <HintPath>..\packages\Microsoft.Extensions.FileProviders.Abstractions.3.1.8\lib\netstandard2.0\Microsoft.Extensions.FileProviders.Abstractions.dll</HintPath>
    </Reference>
    <Reference Include="Microsoft.Extensions.Hosting.Abstractions, Version=3.1.8.0, Culture=neutral, PublicKeyToken=adb9793829ddae60, processorArchitecture=MSIL">
      <HintPath>..\packages\Microsoft.Extensions.Hosting.Abstractions.3.1.8\lib\netstandard2.0\Microsoft.Extensions.Hosting.Abstractions.dll</HintPath>
    </Reference>
    <Reference Include="Microsoft.Extensions.Logging, Version=2.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60, processorArchitecture=MSIL">
      <HintPath>..\packages\Microsoft.Extensions.Logging.2.0.0\lib\netstandard2.0\Microsoft.Extensions.Logging.dll</HintPath>
    </Reference>
    <Reference Include="Microsoft.Extensions.Logging.Abstractions, Version=6.0.0.2, Culture=neutral, PublicKeyToken=adb9793829ddae60, processorArchitecture=MSIL">
      <HintPath>..\packages\Microsoft.Extensions.Logging.Abstractions.6.0.2\lib\net461\Microsoft.Extensions.Logging.Abstractions.dll</HintPath>
    </Reference>
    <Reference Include="Microsoft.Extensions.Options, Version=2.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60, processorArchitecture=MSIL">
      <HintPath>..\packages\Microsoft.Extensions.Options.2.0.0\lib\netstandard2.0\Microsoft.Extensions.Options.dll</HintPath>
    </Reference>
    <Reference Include="Microsoft.Extensions.Primitives, Version=3.1.8.0, Culture=neutral, PublicKeyToken=adb9793829ddae60, processorArchitecture=MSIL">
      <HintPath>..\packages\Microsoft.Extensions.Primitives.3.1.8\lib\netstandard2.0\Microsoft.Extensions.Primitives.dll</HintPath>
    </Reference>
    <Reference Include="Serilog, Version=2.0.0.0, Culture=neutral, PublicKeyToken=24c2f752a8e58a10, processorArchitecture=MSIL">
      <HintPath>..\packages\Serilog.2.10.0\lib\net46\Serilog.dll</HintPath>
    </Reference>
    <Reference Include="Serilog.Extensions.Hosting, Version=2.0.0.0, Culture=neutral, PublicKeyToken=24c2f752a8e58a10, processorArchitecture=MSIL">
      <HintPath>..\packages\Serilog.Extensions.Hosting.5.0.1\lib\netstandard2.0\Serilog.Extensions.Hosting.dll</HintPath>
    </Reference>
    <Reference Include="Serilog.Extensions.Logging, Version=2.0.0.0, Culture=neutral, PublicKeyToken=24c2f752a8e58a10, processorArchitecture=MSIL">
      <HintPath>..\packages\Serilog.Extensions.Logging.3.1.0\lib\netstandard2.0\Serilog.Extensions.Logging.dll</HintPath>
    </Reference>
    <Reference Include="Serilog.Sinks.Console, Version=4.1.0.0, Culture=neutral, PublicKeyToken=24c2f752a8e58a10, processorArchitecture=MSIL">
      <HintPath>..\packages\Serilog.Sinks.Console.4.1.0\lib\net45\Serilog.Sinks.Console.dll</HintPath>
    </Reference>
    <Reference Include="System" />
    <Reference Include="System.Buffers, Version=4.0.3.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51, processorArchitecture=MSIL">
      <HintPath>..\packages\System.Buffers.4.5.1\lib\net461\System.Buffers.dll</HintPath>
    </Reference>
    <Reference Include="System.Memory, Version=4.0.1.1, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51, processorArchitecture=MSIL">
      <HintPath>..\packages\System.Memory.4.5.4\lib\net461\System.Memory.dll</HintPath>
    </Reference>
    <Reference Include="System.Numerics" />
    <Reference Include="System.Numerics.Vectors, Version=4.1.4.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a, processorArchitecture=MSIL">
      <HintPath>..\packages\System.Numerics.Vectors.4.5.0\lib\net46\System.Numerics.Vectors.dll</HintPath>
    </Reference>
    <Reference Include="System.Runtime.CompilerServices.Unsafe, Version=4.0.6.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a, processorArchitecture=MSIL">
      <HintPath>..\packages\System.Runtime.CompilerServices.Unsafe.4.7.1\lib\net461\System.Runtime.CompilerServices.Unsafe.dll</HintPath>
    </Reference>
    <Reference Include="System.Threading.Tasks.Extensions, Version=4.2.0.1, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51, processorArchitecture=MSIL">
      <HintPath>..\packages\System.Threading.Tasks.Extensions.4.5.4\lib\net461\System.Threading.Tasks.Extensions.dll</HintPath>
    </Reference>
  </ItemGroup>
  <ItemGroup>
    <None Include="app.config" />
    <None Include="packages.config" />
  </ItemGroup>
  <Import Project="$(MSBuildToolsPath)\Microsoft.CSharp.targets" />
  <Import Project="..\packages\Microsoft.Extensions.Logging.Abstractions.6.0.2\build\Microsoft.Extensions.Logging.Abstractions.targets" Condition="Exists('..\packages\Microsoft.Extensions.Logging.Abstractions.6.0.2\build\Microsoft.Extensions.Logging.Abstractions.targets')" />
  <Target Name="EnsureNuGetPackageBuildImports" BeforeTargets="PrepareForBuild">
    <PropertyGroup>
      <ErrorText>This project references NuGet package(s) that are missing on this computer. Use NuGet Package Restore to download them.  For more information, see http://go.microsoft.com/fwlink/?LinkID=322105. The missing file is {0}.</ErrorText>
    </PropertyGroup>
    <Error Condition="!Exists('..\packages\Microsoft.Extensions.Logging.Abstractions.6.0.2\build\Microsoft.Extensions.Logging.Abstractions.targets')" Text="$([System.String]::Format('$(ErrorText)', '..\packages\Microsoft.Extensions.Logging.Abstractions.6.0.2\build\Microsoft.Extensions.Logging.Abstractions.targets'))" />
  </Target>
</Project>
```

```xml
<?xml version="1.0" encoding="utf-8"?>
<configuration>
  <runtime>
    <assemblyBinding xmlns="urn:schemas-microsoft-com:asm.v1">
      <dependentAssembly>
        <assemblyIdentity name="Microsoft.Extensions.Logging.Abstractions" publicKeyToken="adb9793829ddae60" culture="neutral" />
        <bindingRedirect oldVersion="0.0.0.0-6.0.0.2" newVersion="6.0.0.2" />
      </dependentAssembly>
      <dependentAssembly>
        <assemblyIdentity name="System.Runtime.CompilerServices.Unsafe" publicKeyToken="b03f5f7f11d50a3a" culture="neutral" />
        <bindingRedirect oldVersion="0.0.0.0-4.0.6.0" newVersion="4.0.6.0" />
      </dependentAssembly>
      <dependentAssembly>
        <assemblyIdentity name="Microsoft.Extensions.Primitives" publicKeyToken="adb9793829ddae60" culture="neutral" />
        <bindingRedirect oldVersion="0.0.0.0-3.1.8.0" newVersion="3.1.8.0" />
      </dependentAssembly>
      <dependentAssembly>
        <assemblyIdentity name="Microsoft.Extensions.DependencyInjection.Abstractions" publicKeyToken="adb9793829ddae60" culture="neutral" />
        <bindingRedirect oldVersion="0.0.0.0-3.1.8.0" newVersion="3.1.8.0" />
      </dependentAssembly>
      <dependentAssembly>
        <assemblyIdentity name="System.Threading.Tasks.Extensions" publicKeyToken="cc7b13ffcd2ddd51" culture="neutral" />
        <bindingRedirect oldVersion="0.0.0.0-4.2.0.1" newVersion="4.2.0.1" />
      </dependentAssembly>
      <dependentAssembly>
        <assemblyIdentity name="Microsoft.Extensions.Configuration.Abstractions" publicKeyToken="adb9793829ddae60" culture="neutral" />
        <bindingRedirect oldVersion="0.0.0.0-3.1.8.0" newVersion="3.1.8.0" />
      </dependentAssembly>
      <dependentAssembly>
        <assemblyIdentity name="Microsoft.Extensions.FileProviders.Abstractions" publicKeyToken="adb9793829ddae60" culture="neutral" />
        <bindingRedirect oldVersion="0.0.0.0-3.1.8.0" newVersion="3.1.8.0" />
      </dependentAssembly>
      <dependentAssembly>
        <assemblyIdentity name="System.Memory" publicKeyToken="cc7b13ffcd2ddd51" culture="neutral" />
        <bindingRedirect oldVersion="0.0.0.0-4.0.1.1" newVersion="4.0.1.1" />
      </dependentAssembly>
    </assemblyBinding>
  </runtime>
</configuration>
```

```xml
<?xml version="1.0" encoding="utf-8"?>
<packages>
  <package id="FluentFTP" version="40.0.0" targetFramework="net472" />
  <package id="Microsoft.Bcl.AsyncInterfaces" version="1.1.1" targetFramework="net472" />
  <package id="Microsoft.Extensions.Configuration.Abstractions" version="3.1.8" targetFramework="net472" />
  <package id="Microsoft.Extensions.DependencyInjection.Abstractions" version="3.1.8" targetFramework="net472" />
  <package id="Microsoft.Extensions.FileProviders.Abstractions" version="3.1.8" targetFramework="net472" />
  <package id="Microsoft.Extensions.Hosting.Abstractions" version="3.1.8" targetFramework="net472" />
  <package id="Microsoft.Extensions.Logging" version="2.0.0" targetFramework="net472" />
  <package id="Microsoft.Extensions.Logging.Abstractions" version="6.0.2" targetFramework="net472" />
  <package id="Microsoft.Extensions.Options" version="2.0.0" targetFramework="net472" />
  <package id="Microsoft.Extensions.Primitives" version="3.1.8" targetFramework="net472" />
  <package id="Serilog" version="2.10.0" targetFramework="net472" />
  <package id="Serilog.Extensions.Hosting" version="5.0.1" targetFramework="net472" />
  <package id="Serilog.Extensions.Logging" version="3.1.0" targetFramework="net472" />
  <package id="Serilog.Sinks.Console" version="4.1.0" targetFramework="net472" />
  <package id="System.Buffers" version="4.5.1" targetFramework="net472" />
  <package id="System.Memory" version="4.5.4" targetFramework="net472" />
  <package id="System.Numerics.Vectors" version="4.5.0" targetFramework="net472" />
  <package id="System.Runtime.CompilerServices.Unsafe" version="4.7.1" targetFramework="net472" />
  <package id="System.Threading.Tasks.Extensions" version="4.5.4" targetFramework="net472" />
</packages>
```

</details>

<details>
<summary>new/SDK style csproj</summary>

```csproj
<Project Sdk="Microsoft.NET.Sdk">

    <PropertyGroup>
        <OutputType>Exe</OutputType>
        <TargetFramework>net472</TargetFramework>
        <ImplicitUsings>enable</ImplicitUsings>
        <LangVersion>10.0</LangVersion>
    </PropertyGroup>

    <ItemGroup>
        <PackageReference Include="FluentFTP" Version="40.0.0" />
        <PackageReference Include="Microsoft.Extensions.Logging.Abstractions" Version="6.0.2" />
        <PackageReference Include="Serilog.Extensions.Hosting" Version="5.0.1" />
        <PackageReference Include="Serilog.Sinks.Console" Version="4.1.0" />
    </ItemGroup>

</Project>
```

</details>

**Comment by robinrodricks** [comment]
Ah ok I knew about it just the term confused me. Yes we should surely switch to it.

**Comment by FanDjango** [comment]
Ok. Using this information about the binding redirects from @jnyrup I have gone and looked at my app.config in the project folder:

I needed to modify it, because the following was not there correctly:
```
      <dependentAssembly>
        <assemblyIdentity name="Microsoft.Extensions.Logging.Abstractions" publicKeyToken="adb9793829ddae60" culture="neutral" />
        <bindingRedirect oldVersion="0.0.0.0-6.0.0.2" newVersion="6.0.0.2" />
      </dependentAssembly>
```

It was, before:
```
<bindingRedirect oldVersion="0.0.0.0-6.0.0.1" newVersion="6.0.0.1" />
```

I had initally installed 6.0.0.1 on this app, some weeks later I was able to update to 6.0.0.2. The app.config was **not** updated to reflect this. For the future I will make a note to myself: This needs to checked always, what a PITA.

I don't understand how this could be, is this not updated by an updated :-) when I updated from 6.0.0.1 to 6.0.0.2?

Anyway once I fixed that, my app, which uses 6.0.0.2 can work with the FluentFTP dll, which is on 2.1.0.

So thats what @Rlooping needs to check next.
